src/middlewares/auth.js

```diff
@@ -16,51 +17,73 @@ function unauthorized(res) {
 }
 
 export default (scopes) => {
+  // For now we only allow 'admin' scope.
+  const validScopes = ['admin', 'client', 'user'].filter(
+    scope => scopes.includes(scope)
+  );
+
+  if (!validScopes.length) {
+    throw new Error(`No valid scope found in "${scopes}"`);
+  }
+
   return (req, res, next) => {
-    const authHeader = req.headers['authorization'];
-    if (!authHeader) {
+    const authHeader = req.headers.authorization;
+    const authQuery = req.query.authorizationToken;
+
+    if (!authHeader && !authQuery) {
       return unauthorized(res);
     }
 
-    const token = authHeader.split('Bearer ')[1];
-    if (!token) {
-      return unauthorized(res);
+    let token;
+    if (authHeader) {
+      const token = authHeader.split('Bearer ')[1];
+      if (!token) {
+        return unauthorized(res);
+      }
+    } else {
+      token = authQuery;
     }
 
     // Because we expect to get tokens signed with different secrets, we first
     // need to get the owner of the token so we can get the appropriate secret.
     const decoded = jwt.decode(token);
 
-    // For now we only allow authenticated requests from the admin user.
-    // When this changes we will have a different secret per sensor and per
-    // user.
-    if (!decoded || !decoded.id || decoded.id !== 'admin') {
+    if (!decoded || !decoded.id || !decoded.scope) {
+      return unauthorized(res);
+    }
+
```

Comment:
note: directly calling `unauthorized` disallows the redirect to `failureUrl`; this will be fixed once we just pass an error to `next()` so that we can handle it (or not) in the next middleware.